### ISSUE TITLE
Strange crash under Visual Studio debugger

### ISSUE THREAD

**Issue by redknightlois** [dotnet/coreclr#19220]
I have been trying to figure out a weird Visual Studio crash when debugging some complex code. Given that I am working with very unsafe code, and the potential of heap corruption is there. Nonetheless, I have verified along the way using breakpoints the heap using Windbg which should at least gave us some 'guarantees' this is not the case. 

The crash happens when the debugger hits a particular line as demonstrated in the video I am attaching. 
[Error.zip](https://github.com/dotnet/coreclr/files/2246904/Error.zip)
Interestingly enough, if you let it run there is no problem. And even if you step-over very close code it doesn't fail either. 

The following is a repro, it ain't small because the entire code is big, but the code-path is very straightforward.
[ConsoleApp2.zip](https://github.com/dotnet/coreclr/files/2246879/ConsoleApp2.zip)

Given what I learned from the issue it doesn' t look like a codegen issue, because it repros on Debug and the assembler code looks just about right both in Release and Debug.  

Disclaimer: I know I dealing with raw memory and the is a potential that I could be messing up; however, I have been trying to rule out for the last 3 days as many possible causes that I could think of in order to avoid filing an issue for a problem of my own doing. But it could be my code.

**Comment by 0xd4d** [comment]
It's related to `value` argument and the debugger calling its `ToString()` method. A workaround is to disable it in the VS settings.

**Comment by AaronRobinsonMSFT** [comment]
@redknightlois I would start by filing an issue via the [VS Developer Community](https://developercommunity.visualstudio.com/spaces/8/index.html). The VS Debugger works closely with the .NET team and if it is a .NET Core issue, it will be routed here.

**Comment by redknightlois** [comment]
@0xd4d Not really, it also happens without the ``.ToString()``. I put it in there just to show that it worked.

**Comment by 0xd4d** [comment]
Verify the args passed to the string ctor (eg. does it take # chars or # bytes).

**Comment by redknightlois** [comment]
I now know from our slack that lately we had more and more serious 'random' crashes all over the place. Everybody suspected the debugger was acting strange, but nobody thought it was our code fault.
